elastic/kibana
# [Console] Update "Copy as cURL" to interpolate variables and strip request-body comments

Fixes https://github.com/elastic/kibana/issues/136786

## Summary

Fixes "Copy as cURL" command for requests with variables and comments.

### Things updated for "Copy as cURL"

- Replace any variables that exist in the URL and body of a request
- Remove any comments that exist in the URL and body of a request
- Catch any syntax errors and show a toast with an error message
- Add unit tests

### Testing

1. Issue any request in Console with variables and comments. For example:

```
GET ${exampleVariable1} // _search
{
  "query": {
    "${exampleVariable2}": {} // match_all
  }
}
```
2. Click 'Copy as cURL"
3. Paste it into the terminal and click enter

The output should result in a successful request
